### PR TITLE
Add remaining GBA Memory Blocks (minus external memory)

### PR DESCRIPTION
**Pull request by kharland**
- Some cleanup up in memory.dart
- Account for blank sections between blocks

### CODE REVIEW

File: gba/lib/src/memory.dart [hunk 1]
```diff
@@ -105,7 +66,7 @@ class MemoryManager {
     bool isBiosProtected(),
   }) {
     return new MemoryManager.fromBuffer(
-      new Uint8List(_totalRamSize).buffer,
+      new Uint8List(0xffffffff).buffer,
```
Comment (matanlurey) [inline gba/lib/src/memory.dart]: No magic numbers, please. Pull out to const.

File: gba/lib/src/memory.dart [hunk 2]
```diff
@@ -119,68 +80,73 @@ class MemoryManager {
...
-      // Video: Masked.
-      video: new BitwiseAndMemoryMask(
-          _maskVideo, new Memory.view(viewVideo.buffer)),
-      // Working: Masked.
-      work: new BitwiseAndMemoryMask(
-        _maskWork,
-        new Memory.view(viewWork.buffer),
-      ),
-    );
+        // Internal: Masked.s
```
Comment (matanlurey) [inline gba/lib/src/memory.dart]: nit: Remove s at end

File: gba/lib/src/memory.dart [hunk 3]
```diff
@@ -235,3 +201,79 @@ class _ProtectedMemory extends Memory with UnwriteableMemory {
...
+
+  MemoryBlock(ByteBuffer buffer, this.start, this.end)
+      : _bytes = buffer.asUint8List(start, end - start);
+
+  Uint8List get bytes => _bytes;
+
+  int get lengthInBytes => _bytes.lengthInBytes;
+}
+
+abstract class MemoryLayout {
```
Comment (matanlurey) [inline gba/lib/src/memory.dart]: Remove class, just make a few top-level functions.

```dart
MemoryBlock biosBlock(ByteBuffer b)
```

File: gba/lib/src/memory.dart [hunk 4]
```diff
@@ -235,3 +201,79 @@ class _ProtectedMemory extends Memory with UnwriteableMemory {
     return super.read32(address);
   }
 }
+
+class MemoryBlock {
```
Comment (matanlurey) [inline gba/lib/src/memory.dart]: Document

File: gba/lib/src/memory.dart [hunk 5]
```diff
@@ -235,3 +201,79 @@ class _ProtectedMemory extends Memory with UnwriteableMemory {
...
+  final int start;
+  final int end;
+  final Uint8List _bytes;
+
+  MemoryBlock(ByteBuffer buffer, this.start, this.end)
+      : _bytes = buffer.asUint8List(start, end - start);
+
+  Uint8List get bytes => _bytes;
```
Comment (matanlurey) [inline gba/lib/src/memory.dart]: What about a `createMemoryView` function that encapsulates:

```dart
new BitwiseAndMemoryMask(end, new Memory.view(_bytes.buffer));
```

Then none of the getters or properties on this need to be public either.

File: gba/lib/src/memory.dart (hunk 1 above)
Comment (kharland) [inline gba/lib/src/memory.dart]: done

File: gba/lib/src/memory.dart (hunk 2 above)
Comment (kharland) [inline gba/lib/src/memory.dart]: done

File: gba/lib/src/memory.dart (hunk 5 above)
Comment (kharland) [inline gba/lib/src/memory.dart]: sg. Made a private static method.

File: gba/lib/src/memory.dart (hunk 4 above)
Comment (kharland) [inline gba/lib/src/memory.dart]: done

File: gba/lib/src/memory.dart (hunk 3 above)
Comment (kharland) [inline gba/lib/src/memory.dart]: done

**Comment by kharland** [comment]
Made requested changes. Will add tests soon

**Comment by kharland** [comment]
Looks like the dev channel build is failing due to this line in package:presubmit https://github.com/matanlurey/presubmit.dart/blob/master/lib/src/plugins/analyzer.dart#L79

Must be a breaking change to the dart analyzer output.

**Comment by matanlurey** [comment]
Looking at failure.

**Comment by matanlurey** [comment]
@kharland Try now - I updated `presubmit` for dev support.

**Pull request closed without merging**